lib/eddystone-beacon-scanner.js

```diff
@@ -24,47 +32,188 @@ EddystoneBeaconScanner.prototype.startScanning = function(allowDuplicates) {
       noble.once('stateChange', startScanningOnPowerOn);
     }
   };
+  
+  var didLoseBeacon = setInterval(this.didLoseBeacon.bind(this), EXIT_GRACE_PERIOD / 2);
 
   startScanningOnPowerOn();
 };
 
 EddystoneBeaconScanner.prototype.stopScanning = function() {
   debug('stopScanning');
+  
   noble.stopScanning();
+  clearInterval(didLoseBeacon);
 };
 
-EddystoneBeaconScanner.prototype.onDiscover = function(peripheral) {
-  debug('onDiscover: %s', peripheral);
+EddystoneBeaconScanner.prototype.didFindBeacon = function(peripheral) {
+  debug('didFindBeacon: %s', peripheral);
+
+  if (this.isEddyStoneBeacon(peripheral) === true) {
 
-  if (this.isUrlBeacon(peripheral)) {
-    var urlBeacon = this.parseUrlBeacon(peripheral);
+    this.cacheBeacon(peripheral);
 
-    debug('onDiscover: urlBeacon = %s', JSON.stringify(urlBeacon));
-    this.emit('discover', urlBeacon);
+    var beaconInfo = this.parseBeacon(peripheral);
+
+    debug('didFindBeacon: Beacon = %s', JSON.stringify(beaconInfo));
+    this.emit('didFindBeacon', beaconInfo);
   }
 };
 
-EddystoneBeaconScanner.prototype.isUrlBeacon = function(peripheral) {
+EddystoneBeaconScanner.prototype.didLoseBeacon = function() {
+  for (var id in inRange) {
+    if (inRange[id].lastSeen < (Date.now() - EXIT_GRACE_PERIOD)) {
+      var peripheral = inRange[id].peripheral;
+      var lostBeacon = this.parseBeacon(peripheral);
```

Comment:
I think is that the TLM frame get's cached with the same peripheral ID as the UID frame and causes things to break. 

One strange thing that I haven't figured out yet is for some even if you ignore caching TLM frames, the TLM data still get's pushed to `inRange[]`.

It's some really odd behavior since the cache functions never fires on the TLM but the data somehow gets pushed.

The code doesn't reflect that now, but you can try and ignore TLM frames but still see that the data gets set on `inRange`